### DISCUSSION TITLE
What's the purpose of Utils.removeCryptographyRestrictions()?

### DISCUSSION THREAD

**Discussion by iscle** [librespot-org/librespot-java#306] [Q&A] [answered]
Hi!

I've noticed this strange function that seems to modify something internal to the JRE to remove some sort of restrictions.

As the title suggests, could you explain what does this do exactly? By looking at the code I don't see anything related to Spotify directly...

Thanks!

**Comment by devgianlu** [comment, marked as answer]
As you said, this is all Java stuff. In particular it is used to remove cryptographic restrictions of the JRE (that I don't even know why are in place). I believe this is needed for the SHA192 used in the Zeroconf blob decoding.

**Reply by iscle** [reply]
Interesting... I did not know Java had restrictions!

Thanks for the answer :)

**Reply by devgianlu** [reply]
Can't remember where I have taken the code from exactly. This should be related: https://stackoverflow.com/questions/6481627/java-security-illegal-key-size-or-default-parameters
